fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java

```diff
@@ -662,6 +667,10 @@ public List<Long> getComputeNodeIds(boolean needAlive) {
     public List<Long> getBackendIds(boolean needAlive) {
         ImmutableMap<Long, Backend> idToBackend = idToBackendRef;
         List<Long> backendIds = Lists.newArrayList(idToBackend.keySet());
+        if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
```

Comment:
we will assume that ware house manages compute node only, so if the caller wants backend, it should only get from idToBackend, should not get from warehouse.